### ISSUE TITLE
How can I make Auto-GPT access local files on my computer, such as PDF and TXT files?

### ISSUE THREAD

**Issue by ChesterZhao** [Significant-Gravitas/Auto-GPT#1930]
I have tried many methods, such as uploading the file to a public repository on GitHub or sharing it on Google Drive, but Auto-GPT still cannot access it.

**Comment by Qoyyuum** [comment]
You can put them in the autogpt_workspace folder

**Comment by ChesterZhao** [comment]
Thank you for your response! I put three PDF files into the autogpt_workspace folder, but its initial reaction is still to go to the browser to search. How can I make it aware that it should go to the autogpt_workspace folder to access the files? Can you give me an example? Thank you very much!

**Comment by Explorergt92** [comment]
@ChesterZhao Set one of your goals to something like "read and review the file <your-file-name> in workspace folder"
and put the file you want it to read in Auto-GPT/auto_gpt_workspace

**Comment by SargeDGz** [comment]
re read the readme specifically the Data Ingestion part

**Comment by finster869** [comment]
I have two files in the auto_gpt_workspace file pb.txt and db.txt.  If I ask the AI in the goals to read and summarize both files it finds them and does so. I have also placed those same files in a subfolder "seed_data" in the auto_gpt_workspace folder. I am using pinecone, so after autogpt is running I am able to successfully run  data_ingestion.py and it tells me the information has been ingested.  How do I let the AI know that db.txt and pb.txt have been ingested to memory?  When it begins processing its commands, the first task it assigns itself is to locate the files pb.txt and db.txt meaning it doesn't know they have been ingested.

**Comment by ChesterZhao** [comment]
How to fix this: SYSTEM:  Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True

**Comment by finster869** [comment]
> How to fix this: SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True

Did you put the file you want read in the auto_gpt_workspace folder?  It needs to be in that folder.

**Comment by ChesterZhao** [comment]
> > How to fix this: SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True
> 
> Did you put the file you want read in the auto_gpt_workspace folder? It needs to be in that folder.

ofcourse~

**Comment by ChesterZhao** [comment]
> > How to fix this: SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True
> 
> Did you put the file you want read in the auto_gpt_workspace folder? It needs to be in that folder.

but it's not working, SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory.

**Comment by finster869** [comment]
> > > How to fix this: SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True
> > 
> > 
> > Did you put the file you want read in the auto_gpt_workspace folder? It needs to be in that folder.
> 
> ofcourse~

Did you indicate in one of the goals for the AI that you want it to read the "xxxxx.txt" file? That is how I do it. If it can't find the file, it then usually automatically searches for itself and finds it.

I can get it to read the text files without issue from the machine. IN contrast, although it gives me a message indicating that the files have been ingested after I run data_ingestion.py, it still searches for them and doesn't seem to realize the information is in its memory.

**Comment by finster869** [comment]
> > > How to fix this: SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True
> > 
> > 
> > Did you put the file you want read in the auto_gpt_workspace folder? It needs to be in that folder.
> 
> but it's not working, SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory.

From what directory are you running autogpt?

**Comment by ChesterZhao** [comment]
> > > > How to fix this: SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory. I already set the EXECUTE_LOCAL_COMMANDS=True
> > > 
> > > 
> > > Did you put the file you want read in the auto_gpt_workspace folder? It needs to be in that folder.
> > 
> > 
> > but it's not working, SYSTEM: Command read_file returned: Error: Attempted to access outside of working directory.
> 
> From what directory are you running autogpt?

yes

**Comment by nayrmot** [comment]
I am getting an error " Command search_files returned: Error: 'WindowsPath' object has no attribute 'is_relative_to'"

How do I resolve this error?

**Comment by willmoura** [comment]
I'm getting this error "SYSTEM:  Command read_file returned: Error: 'utf-8' codec can't decode byte 0xe2 in position 10: invalid continuation byte" How do I resolve it?

**Comment by Nick-Bravante** [comment]
> I'm getting this error "SYSTEM: Command read_file returned: Error: 'utf-8' codec can't decode byte 0xe2 in position 10: invalid continuation byte" How do I resolve it?

I am having this same issue. The system is saying SYSTEM:  Command evaluate_code returned: "Error: The model: `gpt-4` does not exist". Could running 3.5 and not 4 be the issue?

**Comment by magamagaman1** [comment]
Same issue here.  My AI isn't able to read pdf files .  It always gives me :  Command read_file returned: Error: 'utf-8' codec can't decode byte 0xb5 in position 11: invalid start byte .   I already installed PyPDF2 and told it to use it and it still doesn't read them.   My goal for it is to have it read and learn from pdf files.

**Comment by interzone2** [comment]
@magamagaman1 - I think we need to convert PDF to .txt files for this to work.

BUT

Why does autogpt look here for local files: /home/appuser/auto_gpt_workspace/ - this does not correspond to my file structure, there's no appuser directory ?

**Comment by magamagaman1** [comment]
Appuser is the user that is used when you run autogpt in a docker and a
local folder is mapped into the docker.  Converting pdf files to text is
tedious . I have many files that I would like the ai to learn from and I am
sure that we should be able to use something like PyPDF2 but even while I
have installed it, the AI will not use to read the pdf file.
Hope others will see this post and will help get this feature working.
Thanks for your help.

**Comment by edwardliufirm** [comment]
> Same issue here. My AI isn't able to read pdf files . It always gives me : Command read_file returned: Error: 'utf-8' codec can't decode byte 0xb5 in position 11: invalid start byte . I already installed PyPDF2 and told it to use it and it still doesn't read them. My goal for it is to have it read and learn from pdf files.

I went through a same process and it's still not working.

**Comment by interzonez** [comment]
Thanks for the replies. 

I did manage to get someplace with Langchain and memory for querying texts by going through their notebooks hear. It's quite simple. Therefore I'm not sure what is causing the breakage in AutoGPT. Once content has been ingested it should be relatively simple to have GPT take it on board. Of course there are varying levels of complexity. Asking questions of a document is one thing. But I want GPT to go off and perform other tasks WITH this memory activated so that it may use this knowledge to get further along with a problem. Will keep on digging. 

https://python.langchain.com/en/latest/use_cases/summarization.html

**Comment by XChikuX** [comment]
How large can a document be in the workspace folder? 

Can I give it a 700 page book?

**Comment by avaneeshkum** [comment]
@XChikuX yes - for me it hanged, I believe you can integrate a SQLdb or pandas agent.

**Comment by XChikuX** [comment]
@avaneeshkum Do you happen to know what size document would work without the agent?

**Comment by avaneeshkum** [comment]
Last, I used an 800-page book, and it could load it. That was just embeddings and query the doc framework in langchain.

**Comment by shivvem** [comment]
> re read the readme specifically the Data Ingestion part
I have carefully read the instructions. I am using the data_ingestion.py correctly I believe. Yet I get this error, when I try to read my documents placed in auto_gpt_workspace.

 File "C:\Users\shivv\AppData\Local\Programs\Python\Python311\Lib\pathlib.py", line 493, in _parse_args
    a = os.fspath(a)
        ^^^^^^^^^
TypeError: expected str, bytes or os.PathLike object, not NoneType

**Comment by Kumaava** [comment]
> > re read the readme specifically the Data Ingestion part
> 
> I have carefully read the instructions. I am using the data_ingestion.py correctly I believe. Yet I get this error, when I try to read my documents placed in auto_gpt_workspace.
> 
>  File "C:\Users\shivv\AppData\Local\Programs\Python\Python311\Lib\pathlib.py", line 493, in _parse_args
>     a = os.fspath(a)
>         ^^^^^^^^^
> TypeError: expected str, bytes or os.PathLike object, not NoneType
> 
> 

Please keep files in autogpt workspace.

**Comment by shivvem** [comment]
> > > re read the readme specifically the Data Ingestion part
> > 
> > 
> > I have carefully read the instructions. I am using the data_ingestion.py correctly I believe. Yet I get this error, when I try to read my documents placed in auto_gpt_workspace.
> > File "C:\Users\shivv\AppData\Local\Programs\Python\Python311\Lib\pathlib.py", line 493, in _parse_args
> > a = os.fspath(a)
> > ^^^^^^^^^
> > TypeError: expected str, bytes or os.PathLike object, not NoneType
> 
> Please keep files in autogpt workspace.

Sorry, but I am getting the same error over and over again. I have tried putting my files (which I want ingested) at multiple places. None of them work. I tried putting at:
Auto-GPT\autogpt
Auto-GPT\autogpt\auto-gpt-workspace
Auto-GPT\autogpt\workspace

**Comment by glicerico** [comment]
I was having the same problem with a txt file.

TL;DR:
In your `docker-compose.yaml` file, change:
```
    volumes:
      - ./auto_gpt_workspace:/app/auto_gpt_workspace
```
to
```
    volumes:
      - ./auto_gpt_workspace:/app/autogpt/auto_gpt_workspace
```

A bit longer answer:

I noticed that the command that AutoGPT v0.3.0 was proposing was:
```
NEXT ACTION:  COMMAND = read_file ARGUMENTS = {'filename': '/app/autogpt/auto_gpt_workspace/minireport_CMST_CRFAE.txt'}
```
The path looked different from the declared volume in the `docker-compose.yaml` file:
```
    volumes:
      - ./auto_gpt_workspace:/app/auto_gpt_workspace
```

So I added the missing dir, to make sure the directory is mounted where AutoGPT looks for it.
```
    volumes:
      - ./auto_gpt_workspace:/app/autogpt/auto_gpt_workspace

```
And that worked!

**Comment by shivvem** [comment]
> I was having the same problem with a txt file.
> 
> TL;DR: In your `docker-compose.yaml` file, change:
> 
> ```
>     volumes:
>       - ./auto_gpt_workspace:/app/auto_gpt_workspace
> ```
> 
> to
> 
> ```
>     volumes:
>       - ./auto_gpt_workspace:/app/autogpt/auto_gpt_workspace
> ```
> 
> A bit longer answer:
> 
> I noticed that the command that AutoGPT v0.3.0 was proposing was:
> 
> ```
> NEXT ACTION:  COMMAND = read_file ARGUMENTS = {'filename': '/app/autogpt/auto_gpt_workspace/minireport_CMST_CRFAE.txt'}
> ```
> 
> The path looked different from the declared volume in the `docker-compose.yaml` file:
> 
> ```
>     volumes:
>       - ./auto_gpt_workspace:/app/auto_gpt_workspace
> ```
> 
> So I added the missing dir, to make sure the directory is mounted where AutoGPT looks for it.
> 
> ```
>     volumes:
>       - ./auto_gpt_workspace:/app/autogpt/auto_gpt_workspace
> ```
> 
> And that worked!

Thanks for your inputs! For me though, I don't use Docker. I have sort of given up on AutoGPT by now. There are too many raw edges to this .. which is fine as this is a product under development.

**Comment by bdaoud90** [comment]
Hey all, I have a question about PDFs. Does anyone's prompts immediately start directing to finding a converter? can it not read the PDFs natively?
